### PR TITLE
update spades to 3.11.0; changed to compile from source.

### PR DESCRIPTION
**Pull request by notestaff**
* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [X] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by notestaff** [comment]
this PR is a small cleanup of #6010 ; made it a separate PR in case the small changes break the recipe.  But #6010 should be fine to merge.

File: recipes/spades/meta.yaml [hunk 1]
```diff
@@ -1,29 +1,49 @@
...
+  url: http://cab.spbu.ru/files/release{{ version }}/SPAdes-{{ version }}.tar.gz
+  sha256: 308aa3e6c5fb00221a311a8d32c5e8030990356ae03002351eac10abb66bad1f
 
 build:
   number: 0
-  skip: True  # [not py2k and not py35]
 
 requirements:
   build:
+    - gcc >=4.8.2  # [not osx]
```
Comment (bgruening) [inline recipes/spades/meta.yaml]: is this pinning really needed?

File: recipes/spades/meta.yaml [hunk 2]
```diff
@@ -1,29 +1,49 @@
...
+    - llvm  # [osx]
+    - cmake >=2.8.12
+    - zlib
```
Comment (bgruening) [inline recipes/spades/meta.yaml]: can you pin zlib as well?

File: recipes/spades/meta.yaml [hunk 3]
```diff
@@ -1,29 +1,49 @@
...
+    - bzip2
```
Comment (bgruening) [inline recipes/spades/meta.yaml]: This one as well please: https://github.com/conda-forge/conda-forge.github.io/blob/0661e2ecf4b213ddf36350ca100a5a820eb17eb7/scripts/pin_the_slow_way.py#L40-L85

File: recipes/spades/meta.yaml (hunk 1 above)
Comment (notestaff) [reply recipes/spades/meta.yaml]: spades docs say it is: http://cab.spbu.ru/files/release3.11.0/manual.html#sec2.3

File: recipes/spades/meta.yaml (hunk 3 above)
Comment (notestaff) [reply recipes/spades/meta.yaml]: in most bioconda-recipes/recipes/*/meta.yaml zlib is not pinned.  same for bzip2.  is that a bug?

File: recipes/spades/meta.yaml (hunk 1 above)
Comment (bgruening) [reply recipes/spades/meta.yaml]: This is very unusual for Bioconda, we only have gcc 4.8.5.

Comment (notestaff) [reply recipes/spades/meta.yaml]: pinning to >=4.8.2 is consistent with that, no?

Comment (bgruening) [reply recipes/spades/meta.yaml]: Yes, just saying that this is very unusual ... If you want to keep it, let's merge this PR.

Comment (notestaff) [reply recipes/spades/meta.yaml]: Reason to keep is, if someone later looks at the recipe as a guide for other packaging (like I do sometimes with Homebrew formulas), it's best if the requirement from spades docs is preserved.

**Comment by druvus** [comment]
thanks @notestaff for fixing it!

**Comment by notestaff** [comment]
Hm, building the final package takes 49 min 26 sec
https://travis-ci.org/bioconda/bioconda-recipes/jobs/275144287
Time limit for travis jobs is 50 min: https://docs.travis-ci.com/user/customizing-the-build/

But that's because the builds for the three python versions are done sequentially.   Maybe worth configuring travis to make them separate jobs?  @bgruening

**Comment by bgruening** [comment]
@notestaff we have an extended build time in Bioconda, so this should not a problem for this package.

File: recipes/spades/meta.yaml (hunk 1 above)
Comment (tseemann) [reply recipes/spades/meta.yaml]: I agree with @notestaff - if there is a real minimum, put it in!

Comment (notestaff) [reply recipes/spades/meta.yaml]: I think @bgruening's concern was, this makes it look like all other bioconda recipes have _no_ minimum, which probably isn't true.

**Pull request merged**